Userland/Libraries/LibJS/Runtime/TypedArrayPrototype.cpp

```diff
@@ -318,4 +320,134 @@ JS_DEFINE_NATIVE_FUNCTION(TypedArrayPrototype::to_string_tag_getter)
     return js_string(vm, static_cast<TypedArrayBase&>(this_object).element_name());
 }
 
+// 23.2.3.23 %TypedArray%.prototype.set ( source [ , offset ] ), https://tc39.es/ecma262/#sec-%typedarray%.prototype.set
+JS_DEFINE_NATIVE_FUNCTION(TypedArrayPrototype::set)
+{
+    auto* typed_array = typed_array_from(vm, global_object);
+    if (!typed_array)
+        return {};
+
+    auto source = vm.argument(0);
+
+    auto target_offset = vm.argument(1).to_integer_or_infinity(global_object);
+    if (vm.exception())
+        return {};
+
+    if (target_offset < 0) {
+        vm.throw_exception<JS::RangeError>(global_object);
```

Comment:
No error message?

Reply:
The spec doesn't specify any, and I couldn't really think of any sensible error
if there's some sensible default format, I'll use it.

Reply:
> The spec doesn't specify any

They never do, but that doesn't mean we shouldn't have one :)

You can pick from the available ones in `ErrorType`, and if none fit and you don't want to get creative, use `FixmeAddAnErrorString` :D (yes, that really exists)

Reply:
The specification never specifies the error message, you have to come up with one yourself :^) 
If you're having trouble coming up with one, something like "offset is out of range" or "offset is out of bounds" will work